cellular_configs/v1alpha1/commoncoreranconfig_types.go

```diff
@@ -0,0 +1,103 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CommonCoreRANStatus defines the observed state of CommonCoreRAN
+type CommonCoreRANConfigStatus struct {
+}
+
+// +kubebuilder:object:root=true
+type CommonCoreRANConfig struct {
+	metav1.TypeMeta   `json:",inline" yaml:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+	Spec   CommonCoreRANConfigSpec   `json:"spec,omitempty" yaml:"spec,omitempty"`
+	Status CommonCoreRANConfigStatus `json:"status,omitempty" yaml:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// CommonCoreRANConfigList contains a list of CommonCoreRANConfigs
+type CommonCoreRANConfigList struct {
+	metav1.TypeMeta `json:",inline" yaml:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+	Items           []CommonCoreRANConfig `json:"items" yaml:"items"`
+}
+
+// CommonCoreRANConfigSpec defines the characteristics of a deployment of a network function
+type CommonCoreRANConfigSpec struct {
+	// plmnInfo defines the list of PLMNs to configure for core NFs
+	PlmnInfo []PlmnInfoConfig `json:"plmnInfo,omitempty" yaml:"plmnInfo,omitempty"`
+	// communicationModel defines the communcation strategy between core components, using NRF or without NRF
+	// +optional
+	CommunicationModel []CommunicationModelConfig `json:"communicationModel,omitempty" yaml:"communicationModel,omitempty"`
+}
+
+// PlmnInfoConfig defines the structure of PLMN
+type PlmnInfoConfig struct {
+	// mcc defines the mobile country code
+	// +kubebuilder:validation:Pattern=`[02-79][0-9][0-9]`
+	Mcc string `json:"mcc" yaml:"mcc"`
+	// mnc defines the mobile network code
+	// +kubebuilder:validation:Pattern=`[0-9][0-9][0-9]|[0-9][0-9]`
+	Mnc string `json:"mnc" yaml:"mnc"`
+	// tac defines the tracking area code
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=16777215
+	Tac int `json:"tac" yaml:"tac"`
+	// NSSAI defines the Network Slice Selection Assistance Information
+	// +optional
+	Nssai []Nssai `json:"nssai,omitempty" yaml:"nssai,omitempty"`
+}
+
+// NSSAI defines the Network Slice Selection Assistance Information
+type Nssai struct {
+	// Sst defines Slice/Service Type
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=255
+	Sst int `json:"sst" yaml:"sst"`
+	// Sd defines Service Differentiator
+	// +optional
+	// +kubebuilder:validation:Pattern=`^[A-Fa-f0-9]{6}$`
+	Sd string `json:"sd,omitempty" yaml:"sd,omitempty"`
+	// dnnInfo defines the Data Network Names information
+	DnnInfo []DnnInfo `json:"dnnInfo,omitempty" yaml:"dnnInfo,omitempty"`
+}
+
+// DNN defines the Data Network Names Information
+type DnnInfo struct {
```

Comment:
so this is a reference and the name need to match the NF Deployment. Right?
If so make this clear and add a NOTE. 
One R2 is out we have to refactor this.

Reply:
Okay I will add that @henderiw thanks